File: 299/main.cpp
#include<iostream>
using namespace std;

int main() {
  int testcases;
  cin >> testcases;
  for (int i = 0; i < testcases; i++) {
    int length;
    cin >> length;
    int nums[length];
    for (int i = 0; i < length; i++) {
      cin >> nums[i];
    }
    int swap_count = 0;
    for (int i = 0; i < length; i++) {
      for (int j = 0; j < length - 1; j++) {
        if (nums[j] > nums[j+1]) {
          swap(nums[j], nums[j+1]);
          swap_count++;
        }
      }
    }
    cout << "Optimal train swapping takes " << swap_count << " swaps." << endl;
  }
  return 0;
}

File: 299/main.py
def main():
    for _ in range(int(input())):
        _ = int(input())
        nums = list(map(int, input().split()))

        swap_count = 0
        for i in range(len(nums)):
            for j in range(len(nums) - 1):
                if nums[j] > nums[j + 1]:
                    nums[j], nums[j + 1] = nums[j + 1], nums[j]
                    swap_count += 1

        print(f"Optimal train swapping takes {swap_count} swaps.")


if __name__ == "__main__":
    main()
